### PR TITLE
Session cookie fix

### PR DESCRIPTION
**Pull request by kmorozov**
Constructor parameter for persisting session cookies for the SessionCookieJar and the FileCookieJar.
Ref: https://github.com/guzzle/guzzle/issues/1153

### CODE REVIEW

File: src/Cookie/FileCookieJar.php [hunk 1]
```diff
@@ -44,8 +49,10 @@ public function save($filename)
         $json = [];
         foreach ($this as $cookie) {
             /** @var SetCookie $cookie */
-            if ($cookie->getExpires() && !$cookie->getDiscard()) {
-                $json[] = $cookie->toArray();
+            if($cookie->getExpires() || $this->storeSessionCookies) {
```
Comment (mtdowling) [inline src/Cookie/FileCookieJar.php]: What do you think about refactoring this duplication of if statements out into a static method on CookieJar? Maybe the method could be `shouldPersist(SetCookie $cookie, $allowSessionCookies)` or something similar?

Comment (kmorozov) [inline src/Cookie/FileCookieJar.php]: That seems reasonable. I'll make the change.

**Comment by mtdowling** [comment]
Nice work!

Can you make sure to limit lines to 80 characters, ensure that braces are surrounded by spaces, and add tests? After that, I'll merge.

**Comment by kmorozov** [comment]
I've added the tests and fixed the formatting. Do you have an Eclipse formatter? I've made the changes manually, but it would be easier if Eclipse did it for me :)

File: src/Cookie/CookieJar.php [hunk 2]
```diff
@@ -75,6 +76,26 @@ public static function getCookieValue($value)
...
+     * 
+     * @param SetCookie $cookie Being evaluated.
+     * @param bool $allowSessionCookies If we should presist session cookies
+     * @return bool
+     */
+    public static function shouldPersist(
+        SetCookie $cookie,
+        $allowSessionCookies = false
+    ) {
+    	if($cookie->getExpires() || $allowSessionCookies) {
```
Comment (mtdowling) [inline src/Cookie/CookieJar.php]: Missing a space here between "if" and "("

File: src/Cookie/FileCookieJar.php [hunk 3]
```diff
@@ -44,7 +50,7 @@ public function save($filename)
         $json = [];
         foreach ($this as $cookie) {
             /** @var SetCookie $cookie */
-            if ($cookie->getExpires() && !$cookie->getDiscard()) {
+            if(CookieJar::shouldPersist($cookie, $this->storeSessionCookies)) {
```
Comment (mtdowling) [inline src/Cookie/FileCookieJar.php]: Missing a space here between "if" and "("

File: src/Cookie/SessionCookieJar.php [hunk 4]
```diff
@@ -36,7 +43,7 @@ public function save()
         $json = [];
         foreach ($this as $cookie) {
             /** @var SetCookie $cookie */
-            if ($cookie->getExpires() && !$cookie->getDiscard()) {
+            if(CookieJar::shouldPersist($cookie, $this->storeSessionCookies)) {
```
Comment (mtdowling) [inline src/Cookie/SessionCookieJar.php]: Missing a space here between "if" and "("

File: tests/Cookie/FileCookieJarTest.php [hunk 5]
```diff
@@ -63,9 +63,19 @@ public function testPersistsToFileFile()
         // Load the cookieJar from the file
         $jar = new FileCookieJar($this->file);
 
-        // Weeds out temporary and session cookies
-        $this->assertEquals(2, count($jar));
+        if($testSaveSessionCookie) {
```
Comment (mtdowling) [inline tests/Cookie/FileCookieJarTest.php]: Missing a space here between "if" and "("

File: tests/Cookie/SessionCookieJarTest.php [hunk 6]
```diff
@@ -67,9 +67,19 @@ public function testPersistsToSession()
         // Load the cookieJar from the file
         $jar = new SessionCookieJar($this->sessionVar);
 
-        // Weeds out temporary and session cookies
-        $this->assertEquals(2, count($jar));
+        if($testSaveSessionCookie) {
```
Comment (mtdowling) [inline tests/Cookie/SessionCookieJarTest.php]: Missing a space here between "if" and "("

File: tests/Cookie/SessionCookieJarTest.php [hunk 7]
```diff
@@ -36,9 +36,9 @@ public function testLoadsFromSession()
         unset($_SESSION[$this->sessionVar]);
     }
 
-    public function testPersistsToSession()
+    public function testPersistsToSession($testSaveSessionCookie = false)
```
Comment (mtdowling) [inline tests/Cookie/SessionCookieJarTest.php]: This should have a `@dataProvider` that invokes the test. See https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.dataProvider

File: tests/Cookie/SessionCookieJarTest.php (hunk 6 above)
Comment (GrahamCampbell) [inline tests/Cookie/SessionCookieJarTest.php]: @mtdowling Maybe you should enable https://styleci.io/ on this repo so you don't have to review cs manually?

File: tests/Cookie/SessionCookieJarTest.php [hunk 8]
```diff
@@ -67,9 +67,19 @@ public function testPersistsToSession()
...
+            $this->assertEquals(2, count($jar));
+        }
+
         unset($jar);
         unset($_SESSION[$this->sessionVar]);
     }
+
+    public function testPersistsToSessionWithSessionCookie()
+    {
+        $this->testPersistsToSession(true);
```
Comment (mtdowling) [inline tests/Cookie/SessionCookieJarTest.php]: This should be removed and instead tested using a data provider. See https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.dataProvider

**Comment by mtdowling** [comment]
Sweet! Thanks!

**Pull request merged**